packages/node/src/login/oidc/refresh/TokenRefresher.spec.ts

```diff
@@ -24,44 +24,279 @@ import {
+
+type AccessJwt = {
+  sub: string;
+  iss: string;
+  aud: string;
+  nbf: number;
+  exp: number;
+  cnf: {
+    jkt: string;
+  };
+};
+
+const mockWebId = (): string => "https://my.webid/";
+
+const mockKeyBoundToken = (): AccessJwt => {
+  return {
+    sub: mockWebId(),
+    iss: mockDefaultIssuerConfig().issuer.toString(),
+    aud: "https://resource.example.org",
+    nbf: 1562262611,
+    exp: 1562266216,
+    cnf: {
+      jkt: mockJwk().kid as string,
+    },
+  };
+};
+
+const mockIdTokenPayload = (): IdTokenClaims => {
+  return {
+    sub: "https://my.webid",
+    iss: "https://my.idp/",
+    aud: "https://resource.example.org",
+    exp: 1662266216,
+    iat: 1462266216,
+  };
+};
+
+const mockDpopTokens = (): TokenSet => {
+  return {
+    access_token: JSON.stringify(mockKeyBoundToken()),
+    id_token: mockIdToken(),
+    token_type: "DPoP",
+    expired: () => false,
+    claims: mockIdTokenPayload,
+  };
+};
+
+const setupOidcClientMock = (tokenSet: TokenSet) => {
+  const { Issuer } = jest.requireMock("openid-client");
+  function clientConstructor() {
+    // this is untyped, which makes TS complain
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    this.refresh = jest.fn().mockResolvedValueOnce(tokenSet);
+  }
+  const mockedIssuer = {
+    metadata: mockDefaultIssuerConfig(),
+    Client: clientConstructor,
+  };
+  Issuer.mockReturnValueOnce(mockedIssuer);
+};
+
+const setupDefaultOidcClientMock = () => {
+  setupOidcClientMock(mockDpopTokens());
+};
+
+const mockRefresherDefaultStorageUtility = () =>
+  mockStorageUtility({
+    "solidClientAuthenticationUser:mySession": {
+      issuer: "https://my.idp",
+      codeVerifier: "some code verifier",
+      redirectUri: "https://my.app/redirect",
+      dpop: "true",
+    },
+  });
 
 describe("TokenRefresher", () => {
   const defaultMocks = {
     storageUtility: StorageUtilityMock,
-    tokenRequester: TokenRequesterMock,
+    issuerConfigFetcher: mockIssuerConfigFetcher(mockDefaultIssuerConfig()),
+    clientRegistrar: mockDefaultClientRegistrar(),
   };
 
   function getTokenRefresher(
     mocks: Partial<typeof defaultMocks> = defaultMocks
   ): TokenRefresher {
     return new TokenRefresher(
       mocks.storageUtility ?? defaultMocks.storageUtility,
-      mocks.tokenRequester ?? defaultMocks.tokenRequester
+      mocks.issuerConfigFetcher ?? defaultMocks.issuerConfigFetcher,
+      mocks.clientRegistrar ?? defaultMocks.clientRegistrar
     );
   }
 
-  it("Refreshes the token properly", async () => {
-    const key = "global";
-    const refreshTokenValue = "refresh token value";
+  it("throws if no oidc issuer can be retrieved from storage", async () => {
```

Comment:
```suggestion
  it("throws if no OIDC issuer can be retrieved from storage", async () => {
```

Reply:
Resolved in 8818797c031a7e51ec3c56e1bf2e195d54ff6dee